lib/bigtable/cache/FetchJob.js

```diff
@@ -0,0 +1,135 @@
+"use strict";
+
+const Promise = require("bluebird");
+const debug = require("debug")("yildiz:fetchjob");
+
+const redisClient = require("./redisClient");
+
+class FetchJob {
+  
+  constructor(options = {}, yildiz, metrics) {
+
+    this.options = options;
+    this.redisClient = redisClient.create(options, metrics);
+    this.yildiz = yildiz;
+
+    // How long the keys are going to expire in redis
+    this.expireInSec = (this.options.fetchJob || {}).expireInSec || 60 * 60 * 72; //72 hours
+
+    // Fetch Interval for job
+    this.fetchIntervalInSec = (this.options.fetchJob || {}).fetchIntervalInSec || 3;
+
+    // How long the cache is stored until it should be fetched again
+    this.fetchLastAccess = (this.options.fetchJob || {}).fetchLastAccess || 180;
+
+    // Batch size for storing the cache
+    this.fetchBatchSize = (this.options.fetchJob || {}).fetchBatchSize || 10;
+
+    // Limit in retrieving the lastAccess nodeId from redis
+    this.limit = (this.options.fetchJob || {}).limit || 20;
+
+    this._init();
```

Comment:
async methods should not be called from the constructor, this will always result in ugly error debugging at a later time. Move ._init() -> .init() and call it publicly with await